scripts/generate_tutorial_readmes.py

```diff
@@ -165,13 +174,20 @@ def write_badge_table(notebooks):
         "| - | --- | ---- |",
     ]
 
-    # Add each row of the table
-    for i, local_path in enumerate(notebooks, 1):
+    # Get ordered list of file names
+    notebook_list = [name for name in notebooks if 'Intro' in name]
+    notebook_list += [name for name in notebooks if 'Tutorial' in name]
+    notebook_list += [name for name in notebooks if 'Outro' in name]
 
+    # Add badges
+    for local_path in notebook_list:
+        notebook_name = local_path.split('_')[-1].split('.ipynb')[0]
+        if 'Tutorial' in notebook_name:
+            notebook_name = f"Tutorial {notebook_name.split('Tutorial')[1]}"
```

Comment:
I stared at these `.split` operations for a while without understanding what they're supposed to do. While the code could be a little cleaner, it's not a big deal; but please add a comment explaining what's happening here.

Reply:
yeah I just wanted to extract the name automatically - do you have a less convoluted solution?

Reply:
meh ... i guess i'd use a regular expression here but that's complicated in its own way.

The cannoincal way to strip a suffix used to be `stripped = string[:-len(suffix)]` but that's ugly in its own way. They added `str.removeprefix` and `str.removesuffix` in Python 3.9, but the CI runs don't use that because colab is stuck on like 3.7.

The code's fine, just needs a comment to explain its purpose.

Reply:
ok done